pkg/apis/samplesoperator/v1alpha1/types.go

```diff
@@ -47,30 +55,26 @@ type SamplesResourceSpec struct {
 type SamplesResourceStatus struct {
 	// Conditions represents the available maintenance status of the sample
 	// imagestreams and templates.
-	Conditions []SamplesResourceCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,8,rep,name=conditions"`
+	Conditions    []ResourceCondition
+	SecretCreated bool
```

Comment:
why isn't this a condition?

Reply:
there actually is a `SecretUpdateComplete` condition, but I found maintaining this separate boolean easier/cleaner looking than traversing the conditions array

if you feel strongly about this little piece of duplicate information, I'll remove and replace the use of this boolean with searching  the condition array 

I'll wait for your confirmation ...

Reply:
I feel kinda strongly about it, yeah :)

I don't want us to have to maintain redundant info.

Reply:
change part of next push